src/Graviton/CoreBundle/Compiler/VersionCompilerPass.php

```
@@ -0,0 +1,101 @@
+<?php
+/** A custom compiler pass class */
+
+namespace Graviton\CoreBundle\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Container;
+
+/**
+ * @author   List of contributors <https://github.com/libgraviton/graviton/graphs/contributors>
+ * @license  http://opensource.org/licenses/gpl-license.php GNU Public License
+ * @link     http://swisscom.ch
+ */
+class VersionCompilerPass implements CompilerPassInterface
+{
+
+    /**
+     * add version numbers of packages to the container
+     *
+     * @param ContainerBuilder $container Container
+     *
+     * @return void
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $container->setParameter(
+            'graviton.core.version.data',
+            $this->getPackageVersions($container->getParameter('kernel.root_dir'))
+        );
+    }
+
+    /**
+     * @param string $rootDir path to root dir
+     *
+     * @return array version numbers of packages
+     */
+    private function getPackageVersions($rootDir)
+    {
+        $versions = array();
+        array_push($versions, $this->getContextVersion($rootDir));
+        $versions = $this->getInstalledPackagesVersion($rootDir, $versions);
+
+        return $versions;
+    }
+
+    /**
+     * returns the version of graviton or wrapper
+     *
+     * @param string $rootDir path to root dir
+     * @return array
+     */
+    private function getContextVersion($rootDir)
+    {
+        if (strpos($rootDir, 'vendor')) {
+            //@todo shell exec needs to be refactored
+            $result = shell_exec('cd ' . escapeshellarg($rootDir) . '/../../../../  && composer show -s --no-ansi');
+        } else {
+            $result = shell_exec('composer show -s --no-ansi');
+        }
+        $lines = explode(PHP_EOL, $result);
+        $wrapper = array();
```

The return value is array so if nothing is found it should maybe throw an exception or at least return an empty array?